Add support for building with CMake

I have the fork here: https://github.com/Cazadorro/au/tree/au_cmake Note currently I set cmake standard to C++20, but that is for debugging build issues, I'll set it back to c++14 when it's ready. 

Basically right now I'm attempting to get this library to work with cmake.  I'm going case by case trying to get this library to work with cmake with different tests.  

Some of these tests straight up would never work under MSVC, for example, using `M_PI`, when `#define _USE_MATH_DEFINES` is not used.  It gave me compile errors, so I had to add code to actually get some tests to compile under MSVC.  I also don't see any MSVC configs with Bazel, which is doubly confusing, I was under the impression the library was tested against msvc based on [this](https://github.com/aurora-opensource/au/actions/workflows/msvc-x64-19-29-30151.yml) from the readme

Now I'm getting issues with `math_test`,  here's the first error dump of my compile output: 

```
C:\Repos\au\au/unit_of_measure.hh(809): note: see reference to class template instantiation 'au::InStandardPackOrder<au::Magnitude<>,au::Magnitude<>>' being compiled
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340): error C2794: 'type': is not a member of any direct or indirect base class of 'std::common_type<T,U>'
        with
        [
            T=au::Quantity<au::Meters,int>,
            U=au::Quantity<au::Centi<au::Kelvins>,int>
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340): note: the template instantiation context (the oldest one first) is
C:\Repos\au\au\math_test.cc(325): note: see reference to function template instantiation 'auto au::max<Unit,T>(au::QuantityPoint<Unit,T>,au::QuantityPoint<Unit,T>)' being compiled
        with
        [
            Unit=au::Meters,
            T=int
        ]
C:\Repos\au\au/math.hh(337): note: see reference to function template instantiation 'const _Ty &std::max<au::QuantityPoint<Unit,T>>(const _Ty &,const _Ty &) noexcept(<expr>)' being compiled
        with
        [
            _Ty=au::QuantityPoint<au::Meters,int>,
            Unit=au::Meters,
            T=int
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\utility(75): note: while processing the default template argument of 'au::QuantityPoint<Unit,T>::QuantityPoint(au::QuantityPoint<OtherUnit,OtherRep>)'
        with
        [
            Unit=au::Celsius,
            T=int
        ]
C:\Repos\au\au/quantity_point.hh(104): note: see reference to alias template instantiation 'au::QuantityPoint<Unit,T>::EnableIfImplicitOkIs<true,OtherUnit,OtherRep>' being compiled
        with
        [
            Unit=au::Celsius,
            T=int,
            OtherUnit=au::Meters,
            OtherRep=int
        ]
C:\Repos\au\au/quantity_point.hh(88): note: see reference to function template instantiation 'bool au::QuantityPoint<Unit,T>::should_enable_implicit_construction_from<OtherUnit,OtherRep>(void)' being compiled
        with
        [
            Unit=au::Celsius,
            T=int,
            OtherUnit=au::Meters,
            OtherRep=int
        ]
C:\Repos\au\au/quantity_point.hh(78): note: see reference to function template instantiation 'auto au::operator +<au::Meters,au::Centi<U>,int,int>(au::Quantity<au::Meters,int>,au::Quantity<au::Centi<U>,int>)' being compiled
        with
        [
            U=au::Kelvins
        ]
C:\Repos\au\au/quantity.hh(621): note: see reference to function template instantiation 'auto au::detail::using_common_type<au::Quantity<au::Meters,int>,au::Quantity<au::Centi<U>,int>,au::detail::Plus>(T,au::Quantity<au::Centi<U>,int>,Func)' being compiled
        with
        [
            U=au::Kelvins,
            T=au::Quantity<au::Meters,int>,
            Func=au::detail::Plus
        ]
C:\Repos\au\au/quantity.hh(583): note: see reference to alias template instantiation 'std::common_type_t<T,U>' being compiled
        with
        [
            T=au::Quantity<au::Meters,int>,
            U=au::Quantity<au::Centi<au::Kelvins>,int>
        ]
C:\Repos\au\au/quantity.hh(583): error C2938: 'std::common_type_t' : Failed to specialize alias template
C:\Repos\au\au/quantity.hh(585): error C2057: expected constant expression
C:\Repos\au\au/quantity.hh(447): error C2668: 'au::Quantity<au::Meters,int>::as': ambiguous call to overloaded function
C:\Repos\au\au/quantity.hh(156): note: could be 'auto 
/

```


Entire version at end. 


I'm not sure what the deal is here, these compile errors imply there's some sort of specialization that isn't happening?   If I was missing a header, it should have already complained about it in a separate error, but I'm not seeing that happening.  I know MSVC has some issues with CRTP and constexpr members, but I don't see that happening here either?  And given the above CI tests I would think that this would have been caught if it was a legitimate incompatibility with MSVC?    

This test appears to be causing the issue, but there are some *weird* nonsensical successful compiles and failed compiles in this library with some of the other tests that fail, where I've removed and added tests back in, so I'm not sure how much stock I'd put into this being the issue. 

```
TEST(clamp, QuantityPointTakesOffsetIntoAccount) {
    // Recall that 0 degrees Celsius is 273.15 Kelvins.  We know that `clamp` must take the origin
    // into account for this mixed result.  This means whatever unit we return must be at most 1/20
    // Kelvins, and must evenly divide 1/20 Kelvins.
    constexpr auto celsius_origin = clamp(celsius_pt(0), kelvins_pt(200), kelvins_pt(300));
    ASSERT_TRUE(is_integer(unit_ratio(Kelvins{} / mag<20>(), decltype(celsius_origin)::unit)));
    EXPECT_EQ(celsius_origin, centi(kelvins_pt)(273'15));
}
```

in particular, if this line remains in the code, and everything below is commented out, the tests fails, 

`constexpr auto celsius_origin = clamp(celsius_pt(0), kelvins_pt(200), kelvins_pt(300));`

However, when *everything above is removed*, the test compiles fine... and passes... if QuantityPointConsistentWithStdClampWhenTypesAreIdentical and if QuantityPointProducesResultsInCommonUnitOfInputs are the only test remove, the test also passes.  If either of those two are not commented out, it doesn't work.  If I remove QuantityPointTakesOffsetIntoAccount and comment everything below, things work. 

This has already taken more time than I anticipated, I don't think I'll be able to swing trying to *also* get this tested under clang or gcc right now, I was hoping I could get some help identifying what the issue is. 

<details>
<summary>Entire log for math_test</summary>

```
====================[ Build | math_test | Debug ]===============================
"C:\Program Files\JetBrains\CLion 2022.3\bin\cmake\win\x64\bin\cmake.exe" --build C:\Repos\au\cmake-build-debug --target math_test -j 30
[1/2] Building CXX object au\CMakeFiles\math_test.dir\math_test.cc.obj
FAILED: au/CMakeFiles/math_test.dir/math_test.cc.obj 
C:\PROGRA~2\MICROS~2\2022\BUILDT~1\VC\Tools\MSVC\1438~1.331\bin\Hostx86\x64\cl.exe  /nologo /TP  -IC:\Repos\au -IC:\Repos\au\au -external:IC:\Repos\au\cmake-build-debug\_deps\googletest-src\googletest\include -external:IC:\Repos\au\cmake-build-debug\_deps\googletest-src\googletest -external:IC:\Repos\au\cmake-build-debug\_deps\googletest-src\googlemock\include -external:IC:\Repos\au\cmake-build-debug\_deps\googletest-src\googlemock -external:W0 /DWIN32 /D_WINDOWS /EHsc /Ob0 /Od /RTC1 -std:c++20 -MDd -Zi /permissive /W4 /w14640 /w14242 /w14254 /w14263 /w14265 /w14287 /we4289 /w14296 /w14311 /w14545 /w14546 /w14547 /w14549 /w14555 /w14619 /w14826 /w14905 /w14906 /w14928 /Zc:__cplusplus /DEBUG /showIncludes /Foau\CMakeFiles\math_test.dir\math_test.cc.obj /Fdau\CMakeFiles\math_test.dir\ /FS -c C:\Repos\au\au\math_test.cc
C:\Repos\au\au/utility/string_constant.hh(176): warning C4127: conditional expression is constant
C:\Repos\au\au/utility/string_constant.hh(176): note: consider using 'if constexpr' statement instead
C:\Repos\au\au/utility/string_constant.hh(201): note: while evaluating constexpr function 'au::detail::StringConstant<0>::join'
C:\Repos\au\au/utility/string_constant.hh(196): note: while evaluating constexpr function 'au::detail::join_by'
C:\Repos\au\au/unit_of_measure.hh(83): note: while evaluating constexpr function 'au::detail::concatenate'
C:\Repos\au\au/utility/string_constant.hh(176): note: the template instantiation context (the oldest one first) is
C:\Repos\au\au/units/celsius.hh(35): note: see reference to function template instantiation 'auto au::PrefixApplier<au::Centi>::operator ()<au::Kelvins>(au::QuantityMaker<au::Kelvins>) const' being compiled
C:\Repos\au\au/prefix.hh(35): note: see reference to class template instantiation 'au::QuantityMaker<au::Centi<U>>' being compiled
        with
        [
            U=au::Kelvins
        ]
C:\Repos\au\au/quantity.hh(483): note: see reference to class template instantiation 'au::Centi<U>' being compiled
        with
        [
            U=au::Kelvins
        ]
C:\Repos\au\au/prefix.hh(168): note: see reference to alias template instantiation 'au::detail::ExtendedLabel<1,U>' being compiled
        with
        [
            U=au::Kelvins
        ]
C:\Repos\au\au/unit_of_measure.hh(83): note: see reference to function template instantiation 'au::detail::StringConstant<1> au::detail::concatenate<char[2]>(const char (&)[2])' being compiled
C:\Repos\au\au/utility/string_constant.hh(196): note: see reference to function template instantiation 'au::detail::StringConstant<1> au::detail::join_by<char[1],char[2]>(const SepT (&),const char (&)[2])' being compiled
        with
        [
            SepT=char [1]
        ]
C:\Repos\au\au/utility/string_constant.hh(201): note: see reference to function template instantiation 'au::detail::StringConstant<1> au::detail::StringConstant<0>::join<1>(const au::detail::StringConstant<1> &) const' being compiled
C:\Repos\au\au/utility/string_constant.hh(144): note: see reference to function template instantiation 'void au::detail::StringConstant<0>::join_impl<1,>(char *,const au::detail::StringConstant<1> &) const' being compiled
C:\Repos\au\au/magnitude.hh(412): warning C4127: conditional expression is constant
C:\Repos\au\au/magnitude.hh(412): note: consider using 'if constexpr' statement instead
C:\Repos\au\au/magnitude.hh(412): note: the template instantiation context (the oldest one first) is
C:\Repos\au\au\math_test.cc(94): note: see reference to function template instantiation 'auto au::clamp<au::Kilo<U>,au::Milli<U>,au::Meters,int,int,int>(au::Quantity<au::Kilo<U>,int>,au::Quantity<au::Milli<U>,int>,au::Quantity<au::Meters,int>)' being compiled
        with
        [
            U=au::Meters
        ]
C:\Repos\au\au/math.hh(158): note: while processing the default template argument of 'au::Quantity<au::Milli<U>,int>::Quantity(au::Quantity<OtherUnit,OtherRep>)'
        with
        [
            U=au::Meters
        ]
C:\Repos\au\au/quantity.hh(120): note: see reference to alias template instantiation 'au::Quantity<au::Milli<U>,int>::EnableIfImplicitOkIs<true,OtherUnit,OtherRep>' being compiled
        with
        [
            U=au::Meters,
            OtherUnit=au::Kilo<au::Meters>,
            OtherRep=int
        ]
C:\Repos\au\au/quantity.hh(120): note: see reference to class template instantiation 'au::stdx::conjunction<au::HasSameDimension<U2,T>,au::stdx::disjunction<au::detail::CoreImplicitConversionPolicyImpl<Rep,au::Magnitude<au::Pow<B,6>,au::Pow<au::Prime<5>,6>>,OtherRep>,au::detail::PermitAsCarveOutForIntegerPromotion<Rep,au::Magnitude<au::Pow<B,6>,au::Pow<au::Prime<5>,6>>,OtherRep>>>' being compiled
        with
        [
            U2=au::Milli<au::Meters>,
            T=au::Kilo<au::Meters>,
            Rep=int,
            B=au::Prime<2>,
            OtherRep=int
        ]
C:\Repos\au\au/stdx/type_traits.hh(38): note: see reference to class template instantiation 'au::stdx::conjunction<au::stdx::disjunction<au::detail::CoreImplicitConversionPolicyImpl<Rep,au::Magnitude<au::Pow<B,6>,au::Pow<au::Prime<5>,6>>,OtherRep>,au::detail::PermitAsCarveOutForIntegerPromotion<Rep,au::Magnitude<au::Pow<B,6>,au::Pow<au::Prime<5>,6>>,OtherRep>>>' being compiled
        with
        [
            Rep=int,
            B=au::Prime<2>,
            OtherRep=int
        ]
C:\Repos\au\au/stdx/type_traits.hh(36): note: see reference to class template instantiation 'au::stdx::disjunction<au::detail::CoreImplicitConversionPolicyImpl<Rep,au::Magnitude<au::Pow<B,6>,au::Pow<au::Prime<5>,6>>,OtherRep>,au::detail::PermitAsCarveOutForIntegerPromotion<Rep,au::Magnitude<au::Pow<B,6>,au::Pow<au::Prime<5>,6>>,OtherRep>>' being compiled
        with
        [
            Rep=int,
            B=au::Prime<2>,
            OtherRep=int
        ]
C:\Repos\au\au/stdx/type_traits.hh(46): note: see reference to class template instantiation 'au::detail::CoreImplicitConversionPolicyImpl<Rep,au::Magnitude<au::Pow<B,6>,au::Pow<au::Prime<5>,6>>,OtherRep>' being compiled
        with
        [
            Rep=int,
            B=au::Prime<2>,
            OtherRep=int
        ]
C:\Repos\au\au/conversion_policy.hh(56): note: see reference to class template instantiation 'au::stdx::disjunction<std::is_floating_point<_Rep>,au::stdx::conjunction<std::is_integral<int>,au::IsInteger<ScaleFactor>,au::detail::CanScaleThresholdWithoutOverflow<Rep,ScaleFactor>>>' being compiled
        with
        [
            _Rep=int,
            ScaleFactor=au::Magnitude<au::Pow<au::Prime<2>,6>,au::Pow<au::Prime<5>,6>>,
            Rep=int
        ]
C:\Repos\au\au/stdx/type_traits.hh(46): note: see reference to class template instantiation 'au::stdx::disjunction<au::stdx::conjunction<std::is_integral<int>,au::IsInteger<ScaleFactor>,au::detail::CanScaleThresholdWithoutOverflow<Rep,ScaleFactor>>>' being compiled
        with
        [
            ScaleFactor=au::Magnitude<au::Pow<au::Prime<2>,6>,au::Pow<au::Prime<5>,6>>,
            Rep=int
        ]
C:\Repos\au\au/stdx/type_traits.hh(44): note: see reference to class template instantiation 'au::stdx::conjunction<std::is_integral<int>,au::IsInteger<ScaleFactor>,au::detail::CanScaleThresholdWithoutOverflow<Rep,ScaleFactor>>' being compiled
        with
        [
            ScaleFactor=au::Magnitude<au::Pow<au::Prime<2>,6>,au::Pow<au::Prime<5>,6>>,
            Rep=int
        ]
C:\Repos\au\au/stdx/type_traits.hh(38): note: see reference to class template instantiation 'au::stdx::conjunction<au::IsInteger<ScaleFactor>,au::detail::CanScaleThresholdWithoutOverflow<Rep,ScaleFactor>>' being compiled
        with
        [
            ScaleFactor=au::Magnitude<au::Pow<au::Prime<2>,6>,au::Pow<au::Prime<5>,6>>,
            Rep=int
        ]
C:\Repos\au\au/stdx/type_traits.hh(38): note: see reference to class template instantiation 'au::stdx::conjunction<au::detail::CanScaleThresholdWithoutOverflow<Rep,ScaleFactor>>' being compiled
        with
        [
            Rep=int,
            ScaleFactor=au::Magnitude<au::Pow<au::Prime<2>,6>,au::Pow<au::Prime<5>,6>>
        ]
C:\Repos\au\au/stdx/type_traits.hh(36): note: see reference to class template instantiation 'au::detail::CanScaleThresholdWithoutOverflow<Rep,ScaleFactor>' being compiled
        with
        [
            Rep=int,
            ScaleFactor=au::Magnitude<au::Pow<au::Prime<2>,6>,au::Pow<au::Prime<5>,6>>
        ]
C:\Repos\au\au/conversion_policy.hh(48): note: see reference to function template instantiation 'bool au::can_scale_without_overflow<Rep,au::Pow<B,6>,au::Pow<au::Prime<5>,6>>(au::Magnitude<au::Pow<B,6>,au::Pow<au::Prime<5>,6>>,Rep)' being compiled
        with
        [
            Rep=int,
            B=au::Prime<2>
        ]
C:\Repos\au\au/conversion_policy.hh(30): note: see reference to function template instantiation 'T au::get_value<double,au::Pow<B,6>,au::Pow<au::Prime<5>,6>>(au::Magnitude<au::Pow<B,6>,au::Pow<au::Prime<5>,6>>)' being compiled
        with
        [
            T=double,
            B=au::Prime<2>
        ]
C:\Repos\au\au/magnitude.hh(523): note: see reference to function template instantiation 'au::detail::MagRepresentationOrError<T> au::detail::get_value_result<T,au::Pow<B,6>,au::Pow<au::Prime<5>,6>>(au::Magnitude<au::Pow<B,6>,au::Pow<au::Prime<5>,6>>)' being compiled
        with
        [
            T=double,
            B=au::Prime<2>
        ]
C:\Repos\au\au/magnitude.hh(494): note: see reference to function template instantiation 'au::detail::MagRepresentationOrError<long double> au::detail::base_power_value<T,6,1,uintmax_t>(B)' being compiled
        with
        [
            T=double,
            B=uintmax_t
        ]
C:\Repos\au\au/packs.hh(348): warning C4296: '>': expression is always false
C:\Repos\au\au/packs.hh(348): note: the template instantiation context (the oldest one first) is
C:\Repos\au\au\math_test.cc(133): note: see reference to function template instantiation 'auto au::clamp<Unit,au::Kelvins,au::Kelvins,T,T,T>(au::QuantityPoint<Unit,T>,au::QuantityPoint<au::Kelvins,T>,au::QuantityPoint<au::Kelvins,T>)' being compiled
        with
        [
            Unit=au::Celsius,
            T=int
        ]
C:\Repos\au\au/math.hh(186): note: see reference to alias template instantiation 'au::CommonPointUnitT<UV,ULo,UHi>' being compiled
        with
        [
            UV=au::Celsius,
            ULo=au::Kelvins,
            UHi=au::Kelvins
        ]
C:\Repos\au\au/math.hh(186): note: see reference to class template instantiation 'au::ComputeCommonPointUnit<UV,ULo,UHi>' being compiled
        with
        [
            UV=au::Celsius,
            ULo=au::Kelvins,
            UHi=au::Kelvins
        ]
C:\Repos\au\au/unit_of_measure.hh(644): note: see reference to alias template instantiation 'au::ComputeCommonPointUnitImpl<UV,ULo,UHi>' being compiled
        with
        [
            UV=au::Celsius,
            ULo=au::Kelvins,
            UHi=au::Kelvins
        ]
C:\Repos\au\au/unit_of_measure.hh(640): note: see reference to alias template instantiation 'au::FlatDedupedTypeListT<au::CommonPointUnit,UV,ULo,UHi>' being compiled
        with
        [
            UV=au::Celsius,
            ULo=au::Kelvins,
            UHi=au::Kelvins
        ]
C:\Repos\au\au/unit_of_measure.hh(640): note: see reference to class template instantiation 'au::FlatDedupedTypeList<au::CommonPointUnit,au::CommonPointUnit<T>,au::CommonPointUnit<au::Kelvins>,au::CommonPointUnit<au::Kelvins>>' being compiled
        with
        [
            T=au::Celsius
        ]
C:\Repos\au\au/packs.hh(418): note: see reference to alias template instantiation 'au::FlatDedupedTypeListT<List,L1,L2>' being compiled
        with
        [
            List=au::CommonPointUnit,
            L1=au::CommonPointUnit<au::Celsius>,
            L2=au::CommonPointUnit<au::Kelvins>
        ]
C:\Repos\au\au/packs.hh(418): note: see reference to class template instantiation 'au::FlatDedupedTypeList<au::CommonPointUnit,au::CommonPointUnit<T>,au::CommonPointUnit<au::Kelvins>>' being compiled
        with
        [
            T=au::Celsius
        ]
C:\Repos\au\au/packs.hh(397): note: see reference to class template instantiation 'au::InOrderFor<List,T,H>' being compiled
        with
        [
            List=au::CommonPointUnit,
            T=au::Celsius,
            H=au::Kelvins
        ]
C:\Repos\au\au/unit_of_measure.hh(637): note: see reference to class template instantiation 'au::InOrderFor<au::UnitProduct,A,B>' being compiled
        with
        [
            A=au::Celsius,
            B=au::Kelvins
        ]
C:\Repos\au\au/unit_of_measure.hh(860): note: see reference to class template instantiation 'au::LexicographicTotalOrdering<A,B,au::detail::OrderByUnitAvoidance,au::detail::OrderByDim,au::detail::OrderByMag,au::detail::OrderByOrigin,au::detail::OrderAsUnitProduct>' being compiled
        with
        [
            A=au::Celsius,
            B=au::Kelvins
        ]
C:\Repos\au\au/packs.hh(303): note: see reference to class template instantiation 'au::LexicographicTotalOrdering<A,B,au::detail::OrderByDim,au::detail::OrderByMag,au::detail::OrderByOrigin,au::detail::OrderAsUnitProduct>' being compiled
        with
        [
            A=au::Celsius,
            B=au::Kelvins
        ]
C:\Repos\au\au/packs.hh(303): note: see reference to class template instantiation 'au::LexicographicTotalOrdering<A,B,au::detail::OrderByMag,au::detail::OrderByOrigin,au::detail::OrderAsUnitProduct>' being compiled
        with
        [
            A=au::Celsius,
            B=au::Kelvins
        ]
C:\Repos\au\au/packs.hh(318): note: see reference to class template instantiation 'au::detail::OrderByMag<A,B>' being compiled
        with
        [
            A=au::Celsius,
            B=au::Kelvins
        ]
C:\Repos\au\au/unit_of_measure.hh(809): note: see reference to class template instantiation 'au::InStandardPackOrder<au::Magnitude<>,au::Magnitude<>>' being compiled
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340): error C2794: 'type': is not a member of any direct or indirect base class of 'std::common_type<T,U>'
        with
        [
            T=au::Quantity<au::Meters,int>,
            U=au::Quantity<au::Centi<au::Kelvins>,int>
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340): note: the template instantiation context (the oldest one first) is
C:\Repos\au\au\math_test.cc(325): note: see reference to function template instantiation 'auto au::max<Unit,T>(au::QuantityPoint<Unit,T>,au::QuantityPoint<Unit,T>)' being compiled
        with
        [
            Unit=au::Meters,
            T=int
        ]
C:\Repos\au\au/math.hh(337): note: see reference to function template instantiation 'const _Ty &std::max<au::QuantityPoint<Unit,T>>(const _Ty &,const _Ty &) noexcept(<expr>)' being compiled
        with
        [
            _Ty=au::QuantityPoint<au::Meters,int>,
            Unit=au::Meters,
            T=int
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\utility(75): note: while processing the default template argument of 'au::QuantityPoint<Unit,T>::QuantityPoint(au::QuantityPoint<OtherUnit,OtherRep>)'
        with
        [
            Unit=au::Celsius,
            T=int
        ]
C:\Repos\au\au/quantity_point.hh(104): note: see reference to alias template instantiation 'au::QuantityPoint<Unit,T>::EnableIfImplicitOkIs<true,OtherUnit,OtherRep>' being compiled
        with
        [
            Unit=au::Celsius,
            T=int,
            OtherUnit=au::Meters,
            OtherRep=int
        ]
C:\Repos\au\au/quantity_point.hh(88): note: see reference to function template instantiation 'bool au::QuantityPoint<Unit,T>::should_enable_implicit_construction_from<OtherUnit,OtherRep>(void)' being compiled
        with
        [
            Unit=au::Celsius,
            T=int,
            OtherUnit=au::Meters,
            OtherRep=int
        ]
C:\Repos\au\au/quantity_point.hh(78): note: see reference to function template instantiation 'auto au::operator +<au::Meters,au::Centi<U>,int,int>(au::Quantity<au::Meters,int>,au::Quantity<au::Centi<U>,int>)' being compiled
        with
        [
            U=au::Kelvins
        ]
C:\Repos\au\au/quantity.hh(621): note: see reference to function template instantiation 'auto au::detail::using_common_type<au::Quantity<au::Meters,int>,au::Quantity<au::Centi<U>,int>,au::detail::Plus>(T,au::Quantity<au::Centi<U>,int>,Func)' being compiled
        with
        [
            U=au::Kelvins,
            T=au::Quantity<au::Meters,int>,
            Func=au::detail::Plus
        ]
C:\Repos\au\au/quantity.hh(583): note: see reference to alias template instantiation 'std::common_type_t<T,U>' being compiled
        with
        [
            T=au::Quantity<au::Meters,int>,
            U=au::Quantity<au::Centi<au::Kelvins>,int>
        ]
C:\Repos\au\au/quantity.hh(583): error C2938: 'std::common_type_t' : Failed to specialize alias template
C:\Repos\au\au/quantity.hh(585): error C2057: expected constant expression
C:\Repos\au\au/quantity.hh(447): error C2668: 'au::Quantity<au::Meters,int>::as': ambiguous call to overloaded function
C:\Repos\au\au/quantity.hh(156): note: could be 'auto au::Quantity<au::Meters,int>::as<NewRep,enable_if<au::IsUnit<AssociatedUnit<NewUnit>::type>::value,void>::type>(NewUnit) const'
        with
        [
            NewRep=TargetUnit::Rep,
            NewUnit=TargetUnit::Rep
        ]
C:\Repos\au\au/quantity.hh(146): note: or       'auto au::Quantity<au::Meters,int>::as<NewRep,Unit,void>(NewUnit) const'
        with
        [
            NewRep=TargetUnit::Rep,
            Unit=au::Meters,
            NewUnit=au::Meters
        ]
C:\Repos\au\au/quantity.hh(447): note: while trying to match the argument list '(Unit)'
        with
        [
            Unit=au::Meters
        ]
C:\Repos\au\au/quantity.hh(447): note: the template instantiation context (the oldest one first) is
C:\Repos\au\au/quantity.hh(588): note: see reference to function template instantiation 'auto au::detail::cast_to_common_type<C,au::Meters,int>(au::Quantity<au::Meters,int>)' being compiled
C:\Repos\au\au/quantity.hh(578): note: see reference to function template instantiation 'auto au::rep_cast<TargetUnit::Rep,au::Meters,int>(au::Quantity<au::Meters,int>)' being compiled
C:\Repos\au\au/quantity.hh(578): error C3779: 'au::rep_cast': a function that returns 'auto' cannot be used before it is defined
C:\Repos\au\au/quantity.hh(446): note: see declaration of 'au::rep_cast'
C:\Repos\au\au/quantity.hh(447): error C2668: 'au::Quantity<au::Centi<U>,int>::as': ambiguous call to overloaded function
        with
        [
            U=au::Kelvins
        ]
C:\Repos\au\au/quantity.hh(156): note: could be 'auto au::Quantity<au::Centi<U>,int>::as<NewRep,enable_if<au::IsUnit<AssociatedUnit<NewUnit>::type>::value,void>::type>(NewUnit) const'
        with
        [
            U=au::Kelvins,
            NewRep=TargetUnit::Rep,
            NewUnit=TargetUnit::Rep
        ]
C:\Repos\au\au/quantity.hh(146): note: or       'auto au::Quantity<au::Centi<U>,int>::as<NewRep,Unit,void>(NewUnit) const'
        with
        [
            U=au::Kelvins,
            NewRep=TargetUnit::Rep,
            Unit=au::Centi<au::Kelvins>,
            NewUnit=au::Centi<au::Kelvins>
        ]
C:\Repos\au\au/quantity.hh(447): note: while trying to match the argument list '(Unit)'
        with
        [
            Unit=au::Centi<au::Kelvins>
        ]
C:\Repos\au\au/quantity.hh(447): note: the template instantiation context (the oldest one first) is
C:\Repos\au\au/quantity.hh(588): note: see reference to function template instantiation 'auto au::detail::cast_to_common_type<C,au::Centi<U>,int>(au::Quantity<au::Centi<U>,int>)' being compiled
        with
        [
            U=au::Kelvins
        ]
C:\Repos\au\au/quantity.hh(578): note: see reference to function template instantiation 'auto au::rep_cast<TargetUnit::Rep,au::Centi<U>,int>(au::Quantity<au::Centi<U>,int>)' being compiled
        with
        [
            U=au::Kelvins
        ]
C:\Repos\au\au/quantity.hh(588): error C3889: call to object of class type 'au::detail::Plus': no matching call operator found
C:\Repos\au\au/operators.hh(93): note: could be 'auto au::detail::Plus::operator ()(const T &,const U &) const'
C:\Repos\au\au/quantity.hh(588): note: Failed to specialize function template 'auto au::detail::Plus::operator ()(const T &,const U &) const'
C:\Repos\au\au/quantity.hh(588): note: With the following template arguments:
C:\Repos\au\au/quantity.hh(588): note: 'T=void'
C:\Repos\au\au/quantity.hh(588): note: 'U=void'
C:\Repos\au\au/operators.hh(93): note: you cannot create a reference to 'void'
C:\Repos\au\au/quantity_point.hh(76): error C2062: type 'unknown-type' unexpected
C:\Repos\au\au/quantity_point.hh(79): error C2039: 'value': is not a member of '`global namespace''
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340): error C2794: 'type': is not a member of any direct or indirect base class of 'std::common_type<T,U>'
        with
        [
            T=au::Quantity<au::Meters,int>,
            U=au::Quantity<au::Centi<au::Rankines>,int>
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340): note: the template instantiation context (the oldest one first) is
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\utility(75): note: while processing the default template argument of 'au::QuantityPoint<Unit,T>::QuantityPoint(au::QuantityPoint<OtherUnit,OtherRep>)'
        with
        [
            Unit=au::Fahrenheit,
            T=int
        ]
C:\Repos\au\au/quantity_point.hh(104): note: see reference to alias template instantiation 'au::QuantityPoint<Unit,T>::EnableIfImplicitOkIs<true,OtherUnit,OtherRep>' being compiled
        with
        [
            Unit=au::Fahrenheit,
            T=int,
            OtherUnit=au::Meters,
            OtherRep=int
        ]
C:\Repos\au\au/quantity_point.hh(88): note: see reference to function template instantiation 'bool au::QuantityPoint<Unit,T>::should_enable_implicit_construction_from<OtherUnit,OtherRep>(void)' being compiled
        with
        [
            Unit=au::Fahrenheit,
            T=int,
            OtherUnit=au::Meters,
            OtherRep=int
        ]
C:\Repos\au\au/quantity_point.hh(78): note: see reference to function template instantiation 'auto au::operator +<au::Meters,au::Centi<U>,int,int>(au::Quantity<au::Meters,int>,au::Quantity<au::Centi<U>,int>)' being compiled
        with
        [
            U=au::Rankines
        ]
C:\Repos\au\au/quantity.hh(621): note: see reference to function template instantiation 'auto au::detail::using_common_type<au::Quantity<au::Meters,int>,au::Quantity<au::Centi<U>,int>,au::detail::Plus>(T,au::Quantity<au::Centi<U>,int>,Func)' being compiled
        with
        [
            U=au::Rankines,
            T=au::Quantity<au::Meters,int>,
            Func=au::detail::Plus
        ]
C:\Repos\au\au/quantity.hh(583): note: see reference to alias template instantiation 'std::common_type_t<T,U>' being compiled
        with
        [
            T=au::Quantity<au::Meters,int>,
            U=au::Quantity<au::Centi<au::Rankines>,int>
        ]
C:\Repos\au\au/quantity.hh(447): error C2668: 'au::Quantity<au::Centi<U>,int>::as': ambiguous call to overloaded function
        with
        [
            U=au::Rankines
        ]
C:\Repos\au\au/quantity.hh(156): note: could be 'auto au::Quantity<au::Centi<U>,int>::as<NewRep,enable_if<au::IsUnit<AssociatedUnit<NewUnit>::type>::value,void>::type>(NewUnit) const'
        with
        [
            U=au::Rankines,
            NewRep=TargetUnit::Rep,
            NewUnit=TargetUnit::Rep
        ]
C:\Repos\au\au/quantity.hh(146): note: or       'auto au::Quantity<au::Centi<U>,int>::as<NewRep,Unit,void>(NewUnit) const'
        with
        [
            U=au::Rankines,
            NewRep=TargetUnit::Rep,
            Unit=au::Centi<au::Rankines>,
            NewUnit=au::Centi<au::Rankines>
        ]
C:\Repos\au\au/quantity.hh(447): note: while trying to match the argument list '(Unit)'
        with
        [
            Unit=au::Centi<au::Rankines>
        ]
C:\Repos\au\au/quantity.hh(447): note: the template instantiation context (the oldest one first) is
C:\Repos\au\au/quantity.hh(588): note: see reference to function template instantiation 'auto au::detail::cast_to_common_type<C,au::Centi<U>,int>(au::Quantity<au::Centi<U>,int>)' being compiled
        with
        [
            U=au::Rankines
        ]
C:\Repos\au\au/quantity.hh(578): note: see reference to function template instantiation 'auto au::rep_cast<TargetUnit::Rep,au::Centi<U>,int>(au::Quantity<au::Centi<U>,int>)' being compiled
        with
        [
            U=au::Rankines
        ]
ninja: build stopped: subcommand failed.
```

<\details>

Tested all tests, the following tests failed to even build `math_test`, `prefix_test`, `quantity_point_test`, `quantity_test`, and `unit_test`.  All other tests succeed in building *and* pass. 

* `math_test` fails (shown above)

* prefix_test`, 
* 
```C:\Repos\au\au/magnitude.hh(529): error C2338: static_assert failed: 'Value outside range of destination type'
C:\Repos\au\au/conversion_policy.hh(48): note: while evaluating constexpr function 'au::can_scale_without_overflow'
C:\Repos\au\au/magnitude.hh(532): error C2338: static_assert failed: 'Unknown error occurred'
C:\Repos\au\au/conversion_policy.hh(48): note: while evaluating constexpr function 'au::can_scale_without_overflow'
C:\Repos\au\au/conversion_policy.hh(48): error C2975: 'B': invalid template argument for 'au::stdx::bool_constant', expected compile-time constant expression
C:\Repos\au\au/stdx/type_traits.hh(29): note: see declaration of 'B'
C:\Repos\au\au/magnitude.hh(532): fatal error C1003: error count exceeds 100; stopping compilation

```
*   `quantity_point_test` (with similar error to `math_test`)
*  `quantity_test` with similar issue to `prefix_test`. 
* `units_test` with:
* ```
C:\Repos\au\au/quantity.hh(165): error C2338: static_assert failed: 'Dangerous conversion for integer Rep!  See: https://aurora-opensource.github.io/au/main/troubleshooting/#dangerous-conversion'
C:\Repos\au\au/quantity.hh(165): note: the template instantiation context (the oldest one first) is
C:\Repos\au\cmake-build-debug\_deps\googletest-src\googletest\include\gtest/gtest.h(1398): note: see reference to function template instantiation 'testing::AssertionResult testing::internal::CmpHelperEQ<T1,T2>(const char *,const char *,const T1 &,const T2 &)' being compiled
        with
        [
            T1=au::Quantity<au::Micro<au::Grams>,long>,
            T2=au::Quantity<au::PoundsMass,long>
        ]
C:\Repos\au\cmake-build-debug\_deps\googletest-src\googletest\include\gtest/gtest.h(1379): note: see reference to function template instantiation 'bool au::operator ==<au::Micro<au::Grams>,au::PoundsMass,long,long>(au::Quantity<au::Micro<au::Grams>,long>,au::Quantity<au::PoundsMass,long>)' being compiled
C:\Repos\au\au/quantity.hh(595): note: see reference to function template instantiation 'auto au::detail::using_common_type<au::Quantity<au::Micro<au::Grams>,long>,au::Quantity<au::PoundsMass,long>,au::detail::Equal>(T,U,Func)' being compiled
        with
        [
            T=au::Quantity<au::Micro<au::Grams>,long>,
            U=au::Quantity<au::PoundsMass,long>,
            Func=au::detail::Equal
        ]
C:\Repos\au\au/quantity.hh(588): note: see reference to function template instantiation 'auto au::detail::cast_to_common_type<C,au::PoundsMass,long>(au::Quantity<au::PoundsMass,long>)' being compiled
C:\Repos\au\au/quantity.hh(578): note: see reference to function template instantiation 'auto au::Quantity<au::PoundsMass,long>::as<au::Micro<au::Grams>,void>(NewUnit) const' being compiled
        with
        [
            NewUnit=au::Micro<au::Grams>
        ]
C:\Repos\au\au/magnitude.hh(370): warning C4244: 'initializing': conversion from 'T' to 'long double', possible loss of data
        with
        [
            T=__int64
        ]

```

*

@chiphogg I made the changes you suggested for units and that those tests work.  I investigated more in to prefix_test.cc not working, the same spooky behavior in math_test is seen there. I first eliminated all tests below and including `TEST(SiPrefixes, PrefixAppliersPredefined)`, then the tests ran.  If I ran none of the test casses inside the test, but kept the test, it would also work.  Then I eliminted everything above, and found that after eliminating everything including `TEST(PrefixApplier, ConvertsUnitToCorrespondingPrefixedType)` at least the first test case in the group would work `EXPECT_EQ(quetta(inches)(1), ronna(inches)(1000))`.  Then I moved the test from `TEST(PrefixApplier, ConvertsUnitToCorrespondingPrefixedType)` into `TEST(SiPrefixes, PrefixAppliersPredefined)` and that didn't work (same errors below) Then I shuffled things around and it still wouldn't work. At this point I had all the test cases commented out in the test case, so it effectivley looked like this

```
TEST(SiPrefixes, PrefixAppliersPredefined) {
//    constexpr auto inches = QuantityMaker<Inches>{};
    constexpr auto make_milli = PrefixApplier<Milli>{};

    constexpr auto inches = QuantityMaker<Inches>{};

//    constexpr auto d = inches(2);
//    EXPECT_THAT(d.in(make_milli(inches)), SameTypeAndValue(2'000));
//   EXPECT_THAT(make_milli(inches)(5'777).coerce_in(inches), SameTypeAndValue(5));

    EXPECT_EQ(quetta(inches)(1), ronna(inches)(1000));
// everything else commented below
}
```

Given this worked with out the test cases from the very first test group, I wanted to see if I could at least get *all* the test cases to work if the first test cases weren't uncommented.  But I ran into more weirdness.   The following works

```
    EXPECT_EQ(quetta(inches)(1), ronna(inches)(1000));
    EXPECT_EQ(ronna(inches)(1), yotta(inches)(1000));
    EXPECT_EQ(yotta(inches)(1), zetta(inches)(1000));
//    EXPECT_EQ(zetta(inches)(1), exa(inches)(1000));
   // EXPECT_EQ(exa(inches)(1), peta(inches)(1000));
```

The following *doesn't* work (again same error below)
  
```
    EXPECT_EQ(quetta(inches)(1), ronna(inches)(1000));
    EXPECT_EQ(ronna(inches)(1), yotta(inches)(1000));
    EXPECT_EQ(yotta(inches)(1), zetta(inches)(1000));
    EXPECT_EQ(zetta(inches)(1), exa(inches)(1000));
   // EXPECT_EQ(exa(inches)(1), peta(inches)(1000));
```

And the following *does* work

```
//    EXPECT_EQ(quetta(inches)(1), ronna(inches)(1000));
    EXPECT_EQ(ronna(inches)(1), yotta(inches)(1000));
    EXPECT_EQ(yotta(inches)(1), zetta(inches)(1000));
    EXPECT_EQ(zetta(inches)(1), exa(inches)(1000));
   // EXPECT_EQ(exa(inches)(1), peta(inches)(1000));
```

But the following *doesn't work*
```

//    EXPECT_EQ(quetta(inches)(1), ronna(inches)(1000));
    EXPECT_EQ(ronna(inches)(1), yotta(inches)(1000));
    EXPECT_EQ(yotta(inches)(1), zetta(inches)(1000));
    EXPECT_EQ(zetta(inches)(1), exa(inches)(1000));
    EXPECT_EQ(exa(inches)(1), peta(inches)(1000));
```

and in fact, that last line shown above doesn't work unless *the first two above test cases are disabled*

```
//    EXPECT_EQ(quetta(inches)(1), ronna(inches)(1000));
//    EXPECT_EQ(ronna(inches)(1), yotta(inches)(1000));
    EXPECT_EQ(yotta(inches)(1), zetta(inches)(1000));
    EXPECT_EQ(zetta(inches)(1), exa(inches)(1000));
    EXPECT_EQ(exa(inches)(1), peta(inches)(1000));

```

re-arranging doesn't appears to fix things, if I move it like so and uncomment the first it doesn't compile (with any combination of the first two):

```
    EXPECT_EQ(exa(inches)(1), peta(inches)(1000));
    EXPECT_EQ(quetta(inches)(1), ronna(inches)(1000));
//    EXPECT_EQ(ronna(inches)(1), yotta(inches)(1000));
//    EXPECT_EQ(yotta(inches)(1), zetta(inches)(1000));
//    EXPECT_EQ(zetta(inches)(1), exa(inches)(1000));
```

But will still compile in combination with the two it would compile with before. 

```
    EXPECT_EQ(exa(inches)(1), peta(inches)(1000));
//    EXPECT_EQ(quetta(inches)(1), ronna(inches)(1000));
//    EXPECT_EQ(ronna(inches)(1), yotta(inches)(1000));
    EXPECT_EQ(yotta(inches)(1), zetta(inches)(1000));
    EXPECT_EQ(zetta(inches)(1), exa(inches)(1000));
```
 I'm not sure what the deal is here.  I don't really see how this is GoogleTests fault, since I eliminated all other test groups from this test.  I don't see how it's leaving things behind if that's the problem. 


```

====================[ Build | prefix_test | WindowsDebug ]======================
"C:\Program Files\JetBrains\CLion 2022.3\bin\cmake\win\x64\bin\cmake.exe" --build C:\Repos\au\cmake-build-debug --target prefix_test --config Debug
MSBuild version 17.8.3+195e7f5a3 for .NET Framework

  gmock_main.vcxproj -> C:\Repos\au\cmake-build-debug\lib\Debug\gmock_main.lib
  gtest.vcxproj -> C:\Repos\au\cmake-build-debug\lib\Debug\gtest.lib
  gtest_main.vcxproj -> C:\Repos\au\cmake-build-debug\lib\Debug\gtest_main.lib
  prefix_test.cc
C:\Repos\au\au\utility\string_constant.hh(176,29): warning C4127: conditional expression is constant [C:\Repos\au\cmake-build-debug\au\prefix_test.vcxproj]
  (compiling source file '../../au/prefix_test.cc')
  C:\Repos\au\au\utility\string_constant.hh(201,35):
  while evaluating constexpr function 'au::detail::StringConstant<0>::join'
  C:\Repos\au\au\utility\string_constant.hh(196,12):
  while evaluating constexpr function 'au::detail::join_by'
  C:\Repos\au\au\unit_of_measure.hh(83,49):
  while evaluating constexpr function 'au::detail::concatenate'
  C:\Repos\au\au\utility\string_constant.hh(176,29):
  the template instantiation context (the oldest one first) is
  	C:\Repos\au\au\prefix_test.cc(102,5):
  	see reference to function template instantiation 'auto au::PrefixApplier<au::Ronna>::operator ()<au::Inches>(au::QuantityMaker<au::Inches>) const' being compiled
  	C:\Repos\au\au\prefix.hh(35,40):
  	see reference to class template instantiation 'au::QuantityMaker<au::Ronna<U>>' being compiled
          with
          [
              U=au::Inches
          ]
  	C:\Repos\au\au\quantity.hh(483,38):
  	see reference to class template instantiation 'au::Ronna<U>' being compiled
          with
          [
              U=au::Inches
          ]
  	C:\Repos\au\au\prefix.hh(72,30):
  	see reference to alias template instantiation 'au::detail::ExtendedLabel<1,U>' being compiled
          with
          [
              U=au::Inches
          ]
  	C:\Repos\au\au\unit_of_measure.hh(83,49):
  	see reference to function template instantiation 'au::detail::StringConstant<16> au::detail::concatenate<char[17]>(const char (&)[17])' being compiled
  	C:\Repos\au\au\utility\string_constant.hh(196,12):
  	see reference to function template instantiation 'au::detail::StringConstant<16> au::detail::join_by<char[1],char[17]>(const SepT (&),const char (&)[17])' being compiled
          with
          [
              SepT=char [1]
          ]
  	C:\Repos\au\au\utility\string_constant.hh(201,35):
  	see reference to function template instantiation 'au::detail::StringConstant<16> au::detail::StringConstant<0>::join<16>(const au::detail::StringConstant<16> &) const' being compiled
  	C:\Repos\au\au\utility\string_constant.hh(144,9):
  	see reference to function template instantiation 'void au::detail::StringConstant<0>::join_impl<16,>(char *,const au::detail::StringConstant<16> &) const' being compiled
  
C:\Repos\au\au\magnitude.hh(412,11): warning C4127: conditional expression is constant [C:\Repos\au\cmake-build-debug\au\prefix_test.vcxproj]
  (compiling source file '../../au/prefix_test.cc')
  C:\Repos\au\au\magnitude.hh(412,11):
  the template instantiation context (the oldest one first) is
  	C:\Repos\au\au\prefix_test.cc(102,5):
  	see reference to function template instantiation 'testing::AssertionResult testing::internal::EqHelper::Compare<au::Quantity<au::Ronna<U>,int>,au::Quantity<au::Yotta<U>,int>,0x0>(const char *,const char *,const T1 &,const T2 &)' being compiled
          with
          [
              U=au::Inches,
              T1=au::Quantity<au::Ronna<au::Inches>,int>,
              T2=au::Quantity<au::Yotta<au::Inches>,int>
          ]
  	C:\Repos\au\cmake-build-debug\_deps\googletest-src\googletest\include\gtest\gtest.h(1398,12):
  	see reference to function template instantiation 'testing::AssertionResult testing::internal::CmpHelperEQ<T1,T2>(const char *,const char *,const T1 &,const T2 &)' being compiled
          with
          [
              T1=au::Quantity<au::Ronna<au::Inches>,int>,
              T2=au::Quantity<au::Yotta<au::Inches>,int>
          ]
  	C:\Repos\au\cmake-build-debug\_deps\googletest-src\googletest\include\gtest\gtest.h(1379,7):
  	while processing the default template argument of 'au::Quantity<au::Yotta<U>,int>::Quantity(au::Quantity<OtherUnit,OtherRep>)'
          with
          [
              U=au::Inches
          ]
  	C:\Repos\au\au\quantity.hh(120,33):
  	see reference to alias template instantiation 'au::Quantity<au::Yotta<U>,int>::EnableIfImplicitOkIs<true,OtherUnit,OtherRep>' being compiled
          with
          [
              U=au::Inches,
              OtherUnit=au::Ronna<au::Inches>,
              OtherRep=int
          ]
  	C:\Repos\au\au\quantity.hh(120,1):
  	see reference to class template instantiation 'au::stdx::conjunction<au::HasSameDimension<Unit,OtherUnit>,au::stdx::disjunction<au::detail::CoreImplicitConversionPolicyImpl<Rep,au::Magnitude<au::Pow<B,3>,au::Pow<au::Prime<5>,3>>,OtherRep>,au::detail::PermitAsCarveOutForIntegerPromotion<Rep,au::Magnitude<au::Pow<B,3>,au::Pow<au::Prime<5>,3>>,OtherRep>>>' being compiled
          with
          [
              Unit=au::Yotta<au::Inches>,
              OtherUnit=au::Ronna<au::Inches>,
              Rep=int,
              B=au::Prime<2>,
              OtherRep=int
          ]
  	C:\Repos\au\au\stdx\type_traits.hh(38,38):
  	see reference to class template instantiation 'au::stdx::conjunction<au::stdx::disjunction<au::detail::CoreImplicitConversionPolicyImpl<Rep,au::Magnitude<au::Pow<B,3>,au::Pow<au::Prime<5>,3>>,OtherRep>,au::detail::PermitAsCarveOutForIntegerPromotion<Rep,au::Magnitude<au::Pow<B,3>,au::Pow<au::Prime<5>,3>>,OtherRep>>>' being compiled
          with
          [
              Rep=int,
              B=au::Prime<2>,
              OtherRep=int
          ]
  	C:\Repos\au\au\stdx\type_traits.hh(36,26):
  	see reference to class template instantiation 'au::stdx::disjunction<au::detail::CoreImplicitConversionPolicyImpl<Rep,au::Magnitude<au::Pow<B,3>,au::Pow<au::Prime<5>,3>>,OtherRep>,au::detail::PermitAsCarveOutForIntegerPromotion<Rep,au::Magnitude<au::Pow<B,3>,au::Pow<au::Prime<5>,3>>,OtherRep>>' being compiled
          with
          [
              Rep=int,
              B=au::Prime<2>,
              OtherRep=int
          ]
  	C:\Repos\au\au\stdx\type_traits.hh(46,93):
  	see reference to class template instantiation 'au::detail::CoreImplicitConversionPolicyImpl<Rep,au::Magnitude<au::Pow<B,3>,au::Pow<au::Prime<5>,3>>,OtherRep>' being compiled
          with
          [
              Rep=int,
              B=au::Prime<2>,
              OtherRep=int
          ]
  	C:\Repos\au\au\conversion_policy.hh(56,13):
  	see reference to class template instantiation 'au::stdx::disjunction<std::is_floating_point<_Rep>,au::stdx::conjunction<std::is_integral<SourceRep>,au::IsInteger<ScaleFactor>,au::detail::CanScaleThresholdWithoutOverflow<Rep,ScaleFactor>>>' being compiled
          with
          [
              _Rep=int,
              SourceRep=int,
              ScaleFactor=au::Magnitude<au::Pow<au::Prime<2>,3>,au::Pow<au::Prime<5>,3>>,
              Rep=int
          ]
  	C:\Repos\au\au\stdx\type_traits.hh(46,38):
  	see reference to class template instantiation 'au::stdx::disjunction<au::stdx::conjunction<std::is_integral<SourceRep>,au::IsInteger<ScaleFactor>,au::detail::CanScaleThresholdWithoutOverflow<Rep,ScaleFactor>>>' being compiled
          with
          [
              SourceRep=int,
              ScaleFactor=au::Magnitude<au::Pow<au::Prime<2>,3>,au::Pow<au::Prime<5>,3>>,
              Rep=int
          ]
  	C:\Repos\au\au\stdx\type_traits.hh(44,26):
  	see reference to class template instantiation 'au::stdx::conjunction<std::is_integral<SourceRep>,au::IsInteger<ScaleFactor>,au::detail::CanScaleThresholdWithoutOverflow<Rep,ScaleFactor>>' being compiled
          with
          [
              SourceRep=int,
              ScaleFactor=au::Magnitude<au::Pow<au::Prime<2>,3>,au::Pow<au::Prime<5>,3>>,
              Rep=int
          ]
  	C:\Repos\au\au\stdx\type_traits.hh(38,38):
  	see reference to class template instantiation 'au::stdx::conjunction<au::IsInteger<ScaleFactor>,au::detail::CanScaleThresholdWithoutOverflow<Rep,ScaleFactor>>' being compiled
          with
          [
              ScaleFactor=au::Magnitude<au::Pow<au::Prime<2>,3>,au::Pow<au::Prime<5>,3>>,
              Rep=int
          ]
  	C:\Repos\au\au\stdx\type_traits.hh(38,38):
  	see reference to class template instantiation 'au::stdx::conjunction<au::detail::CanScaleThresholdWithoutOverflow<Rep,ScaleFactor>>' being compiled
          with
          [
              Rep=int,
              ScaleFactor=au::Magnitude<au::Pow<au::Prime<2>,3>,au::Pow<au::Prime<5>,3>>
          ]
  	C:\Repos\au\au\stdx\type_traits.hh(36,26):
  	see reference to class template instantiation 'au::detail::CanScaleThresholdWithoutOverflow<Rep,ScaleFactor>' being compiled
          with
          [
              Rep=int,
              ScaleFactor=au::Magnitude<au::Pow<au::Prime<2>,3>,au::Pow<au::Prime<5>,3>>
          ]
  	C:\Repos\au\au\conversion_policy.hh(48,31):
  	see reference to function template instantiation 'bool au::can_scale_without_overflow<Rep,au::Pow<B,3>,au::Pow<au::Prime<5>,3>>(au::Magnitude<au::Pow<B,3>,au::Pow<au::Prime<5>,3>>,Rep)' being compiled
          with
          [
              Rep=int,
              B=au::Prime<2>
          ]
  	C:\Repos\au\au\conversion_policy.hh(30,9):
  	see reference to function template instantiation 'T au::get_value<double,au::Pow<B,3>,au::Pow<au::Prime<5>,3>>(au::Magnitude<au::Pow<B,3>,au::Pow<au::Prime<5>,3>>)' being compiled
          with
          [
              T=double,
              B=au::Prime<2>
          ]
  	C:\Repos\au\au\magnitude.hh(523,29):
  	see reference to function template instantiation 'au::detail::MagRepresentationOrError<T> au::detail::get_value_result<T,au::Pow<B,3>,au::Pow<au::Prime<5>,3>>(au::Magnitude<au::Pow<B,3>,au::Pow<au::Prime<5>,3>>)' being compiled
          with
          [
              T=double,
              B=au::Prime<2>
          ]
  	C:\Repos\au\au\magnitude.hh(494,18):
  	see reference to function template instantiation 'au::detail::MagRepresentationOrError<long double> au::detail::base_power_value<T,3,1,uintmax_t>(B)' being compiled
          with
          [
              T=double,
              B=uintmax_t
          ]
 
C:\Repos\au\au\magnitude.hh(529,34): error C2338: static_assert failed: 'Value outside range of destination type' [C:\Repos\au\cmake-build-debug\au\prefix_test.vcxproj]
  (compiling source file '../../au/prefix_test.cc')
  C:\Repos\au\au\conversion_policy.hh(48,62):
  while evaluating constexpr function 'au::can_scale_without_overflow'
  
C:\Repos\au\au\magnitude.hh(532,34): error C2338: static_assert failed: 'Unknown error occurred' [C:\Repos\au\cmake-build-debug\au\prefix_test.vcxproj]
  (compiling source file '../../au/prefix_test.cc')
  C:\Repos\au\au\conversion_policy.hh(48,62):
  while evaluating constexpr function 'au::can_scale_without_overflow'
  
C:\Repos\au\au\conversion_policy.hh(48,62): error C2975: 'B': invalid template argument for 'au::stdx::bool_constant', expected compile-time constant expression [C:\Repos\au\cmake-build-debug\au\prefix_test.vcxproj]
  (compiling source file '../../au/prefix_test.cc')
  C:\Repos\au\au\stdx\type_traits.hh(29,16):
  see declaration of 'B'


```

@chiphogg Alright, so I had a thought, does the issue still happen if we use `EXPECT_TRUE` instead of `EXPECT_EQ`?  So I replaced all the `EXPECT_EQ`'s in `TEST(SiPrefixes, PrefixAppliersPredefined)`, so they instead looked like    

 ```
    EXPECT_TRUE(quetta(inches)(1) == ronna(inches)(1000));
    EXPECT_TRUE(ronna(inches)(1) == yotta(inches)(1000));
    EXPECT_TRUE(yotta(inches)(1) == zetta(inches)(1000));
    EXPECT_TRUE(zetta(inches)(1) == exa(inches)(1000));
    EXPECT_TRUE(exa(inches)(1) == peta(inches)(1000));
```

and... it compiled.  So I undid commenting out everything else, and the only other part that failed to compile was `TEST(BinaryPrefixes, PrefixAppliersPredefined)` which also has a bunch of prefix tests.  The funny thing is `EXPECT_EQ` is used in many other places, including the tests above it, `TEST(BinaryPrefixes, HaveCorrectAbsoluteValues)`, where it's used like 

`EXPECT_EQ(unit_ratio(Yobi<Bytes>{}, Bytes{}), pow<8>(mag<1024>()));`

And it doesn't fail to compile. So... I'm not sure what is going on, but this should work on MSVC and GCC/CLang and still result in the same tests.   This makes me think it actually *is* a google test related problem, I'm just not sure how. 

So I got prefix_test.cc working, and everything passed.  Then I tried to see if the same issue was with quantity_test.cc... and sure enough *a single line* (where EXPECT_TRUE is below) had to be changed.

```
TEST(Quantity, HandlesBaseDimensionsWithFractionalExponents) {
    using KiloRootFeet = decltype(root<2>(Mega<Feet>{}));
    constexpr auto x = make_quantity<KiloRootFeet>(5);
    EXPECT_EQ(x.in(root<2>(Feet{})), 5'000);
    EXPECT_TRUE(x * x == mega(feet)(25)); //only changed line in the entire file...
}
```


But it's strange since a similar line in the subsequent function wasn't needed (though it was comparing doubles there)). 
```
TEST(Quantity, HandlesMagnitudesWithFractionalExponents) {
    constexpr auto x = sqrt(kilo(feet))(3.0);

    // We can retrieve the value in the same unit (regardless of the scale's fractional powers).
    EXPECT_EQ(x.in(sqrt(kilo(feet))), 3.0);

    // We can retrieve the value in a *different* unit, which *also* has fractional powers, as long
    // as their *ratio* has no fractional powers.
    EXPECT_EQ(x.in(sqrt(milli(feet))), 3'000.0);

    // We can also retrieve the value in a different unit whose ratio *does* have fractional powers.
    EXPECT_NEAR(x.in(sqrt(feet)), 94.86833, 1e-5);

    // Squaring the fractional base power gives us an exact non-fractional dimension and scale.
    EXPECT_EQ(x * x, kilo(feet)(9.0));
}
```


So quanity_test.cc built.  

Then I went to quantity_point_test, and sure enough another EXPECT_EQ was causing compile failure (and again, one very similar  to it up above didn't have the same issue...)

```
TEST(QuantityPoint, SupportsDirectAccessWithQuantityMakerOfSameUnit) {
    auto p = meters_pt(3);
    ++(p.data_in(meters_pt));
    EXPECT_EQ(p, meters_pt(4));
}
```

changed to 

```
TEST(QuantityPoint, SupportsDirectAccessWithQuantityMakerOfSameUnit) {
    auto p = meters_pt(3);
    ++(p.data_in(meters_pt));
    EXPECT_TRUE(p == meters_pt(4));
}

```

But I ran into an issue with google_test's "matchers" and `EXPECT_THAT`.  and while the "out of order" thing didn't really seem to be happening, at least at the begining, I was seeing strange behavior still, and this time de-sugaring didn't seem to help 

I got the following error:

```
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340,53): error C2794: 'type': is not a member of any direct or indirect base class of 'std::common_type<T,U>' [C:\Repos\au\cmake-build-debug\au\quantity_point_test.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340,53): error C2794:         with [C:\Repos\au\cmake-build-debug\au\quantity_point_test.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340,53): error C2794:         [ [C:\Repos\au\cmake-build-debug\au\quantity_point_test.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340,53): error C2794:             T=au::Quantity<au::Meters,int>, [C:\Repos\au\cmake-build-debug\au\quantity_point_test.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340,53): error C2794:             U=au::Quantity<au::ScaledUnit<au::Kelvins,au::Magnitude<au::Pow<au::Prime<2>,-2>,au::Pow<au::Prime<5>,-1>>>,int> [C:\Repos\au\cmake-build-debug\au\quantity_point_test.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340,53): error C2794:         ] [C:\Repos\au\cmake-build-debug\au\quantity_point_test.vcxproj]
```

When a test did the following:

`EXPECT_THAT(centi(meters_pt)(75).coerce_as(meters_pt), SameTypeAndValue(meters_pt(0)));`

Which made no sense,  Where is the kelvins here?   I kept moving things around, until I ended up with this mess that *finally* passed, but I had to *completely avoid matchers* to do it. 


TEST(QuantityPoint, CanImplicitlyConstructWithNonintegerOffsetIffDestinationIsFloatingPoint) {
    {
        using T1 = QuantityPoint<Celsius, int32_t>;
        using T2 = QuantityPoint<Kelvins, int32_t>;
         auto x = std::is_convertible<T1, T2>::value; // this line normally causes issues with the matchers
//        auto x = (std::is_convertible<QuantityPointI32<Celsius>, QuantityPointI32<Kelvins>>::value);
//        EXPECT_FALSE(x);

        auto v = centi(meters_pt)(75).coerce_as(meters_pt);
        using v_type =decltype(v) ;
//        const ::testing::Matcher<const v_type&> matcher = ::testing::SafeMatcherCast<const v_type&>(SameTypeAndValue(meters_pt(0)));

        // The expected path here is that the matcher should match (i.e. that most
        // tests pass) so optimize for this case.
//        auto q = matcher.Matches(v);
//        auto matcher = ::testing::internal::MakePredicateFormatterFromMatcher(SameTypeAndValue(meters_pt(0)));
//        auto q = matcher("", centi(meters_pt)(75).coerce_as(meters_pt));
//        using g_type =decltype( meters_pt(0));
        EXPECT_TRUE((std::is_same_v<v_type, decltype( meters_pt(0))>) && v == meters_pt(0));
    }

//    EXPECT_TRUE((std::is_convertible<QuantityPointI32<Celsius>, QuantityPointD<Kelvins>>::value));
//
//    // Also works for ints if the destination unit evenly divides both offset and initial diff.
//    EXPECT_TRUE(
//        (std::is_convertible<QuantityPointI32<Celsius>, QuantityPointI32<Milli<Kelvins>>>::value));
}


The *same* error will appear even if I try to get around the GTest macro and manually use the underlying matcher stuff.  Some how the mere existence of  `auto x = std::is_convertible<T1, T2>::value;` with Kelvins is causing issues, I don't see how, but it's as if Celcius is being replaced by Meters later in the code at the *ast* or *compiler level* (or vice versa). My best guess is there is *some* sort of bug in MSVC due to how deep google test goes with template instantiation, specifically because of how matchers work. If I make the types both kelvin it works.   

I think the quick solution is *possibly* manually changing stuff that fails to compile with matchers to use EXPECT_TRUE,  but *only* for MSVC when it is detected (since the test failure errors are just worse with out the matchers used here). 
But this seems like a MSVC bug.  Unfortunately, it's not exactly straight forward to report MSVC bugs, you need a non-anonymous account I think, (and at least here, it seems I can't create any more for this device anymore).  Basically I don't have a way to create an account I'm comfortable with associating with my account here, as they'll need to both look at this conversation and earlier commits to my repo. 

If you can, can you report this to https://developercommunity.visualstudio.com/cpp and say something like:


> "Extensive template usage appears to cause inconsistent compiler behavior, resulting in completely wrong templates being substituted in some instances"
> 
> [severity:It’s more difficult to complete my work]
> See https://github.com/Cazadorro/au/tree/au_cmake_msvc_bug_reproduction and the specific file https://github.com/Cazadorro/au/blob/au_cmake_msvc_bug_reproduction/au/quantity_point_test.cc and the discussion on github https://github.com/aurora-opensource/au/issues/215.  Execute the `quantity_point_test` target to reproduce. 
> 
> See lines 72 through 101, 
> 
> ```
> #define MSVC_BUG_TEST_ENABLE_FAILURE_CASE
> TEST(QuantityPoint, CanImplicitlyConstructWithNonintegerOffsetIffDestinationIsFloatingPoint) {
>     {
>         using T1 = QuantityPoint<Celsius, int32_t>;
>         using T2 = QuantityPoint<Kelvins, int32_t>;
>          auto x = std::is_convertible<T1, T2>::value;
> //        auto x = (std::is_convertible<QuantityPointI32<Celsius>, QuantityPointI32<Kelvins>>::value);
> //        EXPECT_FALSE(x);
> 
>         auto v = centi(meters_pt)(75).coerce_as(meters_pt);
>         using v_type =decltype(v) ;
> 
> #ifdef MSVC_BUG_TEST_ENABLE_FAILURE_CASE
>         const ::testing::Matcher<const v_type&> matcher = ::testing::SafeMatcherCast<const v_type&>(SameTypeAndValue(meters_pt(0)));
> #endif
>         // The expected path here is that the matcher should match (i.e. that most
>         // tests pass) so optimize for this case.
> //        auto q = matcher.Matches(v);
> //        auto matcher = ::testing::internal::MakePredicateFormatterFromMatcher(SameTypeAndValue(meters_pt(0)));
> //        auto q = matcher("", centi(meters_pt)(75).coerce_as(meters_pt));
> //        using g_type =decltype( meters_pt(0));
>         EXPECT_TRUE((std::is_same_v<v_type, decltype( meters_pt(0))>) && v == meters_pt(0));
>     }
> 
> //    EXPECT_TRUE((std::is_convertible<QuantityPointI32<Celsius>, QuantityPointD<Kelvins>>::value));
> //
> //    // Also works for ints if the destination unit evenly divides both offset and initial diff.
> //    EXPECT_TRUE(
> //        (std::is_convertible<QuantityPointI32<Celsius>, QuantityPointI32<Milli<Kelvins>>>::value));
> }
> ```
> 
> With `#define MSVC_BUG_TEST_ENABLE_FAILURE_CASE`, the mere creation of `const ::testing::Matcher<const v_type&> matcher = ::testing::SafeMatcherCast<const v_type&>` seemingly causes the line *above* it to fail          `auto x = std::is_convertible<T1, T2>::value;`, and likewise, if this line is removed compilation will also succeed, despite the error being: 
>   
> ```
> C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340,53): error C2794: 'type': is not a member of any direct or indirect base class of 'std::common_type<T,U>' [C:\Repos\au\cmake-build-debug\au\quantity_point_test.vcxproj]
> C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340,53): error C2794:         with [C:\Repos\au\cmake-build-debug\au\quantity_point_test.vcxproj]
> C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340,53): error C2794:         [ [C:\Repos\au\cmake-build-debug\au\quantity_point_test.vcxproj]
> C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340,53): error C2794:             T=au::Quantity<au::Meters,int>, [C:\Repos\au\cmake-build-debug\au\quantity_point_test.vcxproj]
> C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340,53): error C2794:             U=au::Quantity<au::ScaledUnit<au::Kelvins,au::Magnitude<au::Pow<au::Prime<2>,-2>,au::Pow<au::Prime<5>,-1>>>,int> [C:\Repos\au\cmake-build-debug\au\quantity_point_test.vcxproj]
> C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\type_traits(1340,53): error C2794:         ] [C:\Repos\au\cmake-build-debug\au\quantity_point_test.vcxproj]
> 
> ```
> 
> 
> Due to the inconsistent nature of the bug, this is close to the smallest effective MVCE that can be created.

Now that the core support has landed... could somebody please help me figure out how to use it in a new project?  :sweat_smile: 

Here's what I've got.

`main.cc`:

```cpp
#include "au/au.hh"

int main(int argc, char **argv) { return 0; }
```

`CMakeLists.txt`:

```cmake
cmake_minimum_required(VERSION 3.29)

project(Test)

include(FetchContent)
FetchContent_Declare(
  Au
  GIT_REPOSITORY https://github.com/aurora-opensource/au
  GIT_TAG fc050fa51dc10fd010bcb9db5c52159ceaa0369d
)
FetchContent_MakeAvailable(Au)

add_executable(Test main.cpp)
target_link_libraries(Test PUBLIC Au::au)
```

Here are the commands I'm trying to run:

```sh
cmake -S . -B build
cmake --build build
```

Error:

```
CMake Error at CMakeLists.txt:14 (target_link_libraries):
  Target "CppUnitsCompare" links to:

    Au::au

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done (0.0s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
gmake: *** [Makefile:648: cmake_check_build_system] Error 1
```

Thanks in advance!

--- a/docs/install.md
+++ b/docs/install.md
@@ -258,10 +258,10 @@ There are two ways to include the Au library in your CMake project.
 
 2. Install the library to the system, and use `find_package`.
 
-The reason we recommend `FetchContent` is because it makes it easier for each project to get the
-exact version of Au they need.  `FetchContent` also means you don't need to manually clone the Au
-repo, or build and run the tests.  On the other hand, if you want a single global system-wide
-version of Au, then you can install it to the system, and simply use `find_package`.
+We recommend `FetchContent` because each project can get the exact version of Au that they need, and
+can update it independently of other projects.  `FetchContent` also means you don't need to manually
+clone the Au repo, or build and run the tests.  On the other hand, if you want a single global
+system-wide version of Au, then you can install it to the system, and simply use `find_package`.
 
 In either case, here are the main targets and include files provided by the Au library:
 
@@ -270,8 +270,6 @@ In either case, here are the main targets and include files provided by the Au l
 | `Au::au` | `"au/au.hh"`<br>`"au/units/*.hh"` | Core library functionality.  See [all available units](https://github.com/aurora-opensource/au/tree/main/au/units) |
 | `Au::io` | `"au/io.hh"` | `operator<<` support |
 | `Au::testing` | `"au/testing.hh"` | Utilities for writing googletest tests |
-
-Now, here are the installation instructions.
 
 !!! note
     These instructions are for adding Au to a _project_ that uses CMake, not building Au itself

--- a/docs/install.md
+++ b/docs/install.md
@@ -33,17 +33,19 @@ Here's an overview of the tradeoffs involved.
   <tr>
     <th rowspan=2></th>
     <th colspan=2>Single File</th>
-    <th colspan=2>Full Install</th>
+    <th colspan=3>Full Install</th>
   </tr>
   <tr>
     <td>Pre-built</td>
     <td>Custom</td>
     <td>bazel</td>
-    <td>CMake, conan, vcpkg, ...</td>
+    <td>CMake</td>
+    <td>conan, vcpkg, ...</td>
   </tr>
   <tr>
     <td>Setup time</td>
     <td class="best">~1 min</td>
+    <td class="good">~10 min</td>
     <td class="good">~10 min</td>
     <td class="good">~10 min</td>
     <td class="poor">Not <i>yet</i> supported<br>(use <b>single-file</b> instead for now)</td>
@@ -52,13 +54,13 @@ Here's an overview of the tradeoffs involved.
     <td>Unit selection</td>
     <td class="fair">Base units only<br>(or too many units)</td>
     <td class="good">Any units desired</td>
-    <td colspan=2 class="best">Any units desired, <i>without</i> needing "reinstall"</td>
+    <td colspan=3 class="best">Any units desired, <i>without</i> needing "reinstall"</td>
   </tr>
   <tr>
     <td>Compile time cost</td>
     <td class="good">~10 units</td>
     <td class="good">Very competitive up to a few dozen units</td>
-    <td colspan=2 class="best"><i>Each file</i> only pays for the units it uses</td>
+    <td colspan=3 class="best"><i>Each file</i> only pays for the units it uses</td>
   </tr>
   <tr>
     <td>Flexibility</td>
@@ -66,7 +68,7 @@ Here's an overview of the tradeoffs involved.
       Awkward: would need to download <pre>io.hh</pre> and/or <pre>testing.hh</pre> separately, and
       modify their includes manually
     </td>
-    <td colspan=2 class="best">
+    <td colspan=3 class="best">
       Include I/O, testing utilities, individual units as desired, on a per-file basis
     </td>
   </tr>
@@ -87,6 +89,7 @@ SetupTime -->|No! Just let me start!| UsePreBuilt
 SetupTime -->|Sure| UseCustom
 Usage -->|Ready to use in my project!| BuildSystem
 BuildSystem -->|bazel| UseFullInstall
+BuildSystem -->|CMake| UseFullInstall
 BuildSystem -->|other| UseCustom
 ```
 
@@ -243,38 +246,105 @@ attribute, and include the appropriate files.
 
 | Dependency | Headers provided | Notes |
 |------------|------------------|-------|
-| `@au//au` | `"au/au.hh"`<br>`"au/units.*.hh"` | Core library functionality.  See [all available units](https://github.com/aurora-opensource/au/tree/main/au/units) |
+| `@au//au` | `"au/au.hh"`<br>`"au/units/*.hh"` | Core library functionality.  See [all available units](https://github.com/aurora-opensource/au/tree/main/au/units) |
 | `@au//au:io` | `"au/io.hh"` | `operator<<` support |
-| `@au//au:testing` | `"au/testing.hh"` | Utilities for testing<br>_Note:_ `testonly = True` |
+| `@au//au:testing` | `"au/testing.hh"` | Utilities for writing googletest tests<br>_Note:_ `testonly = True` |
 
 #### CMake
 
-CMake support is still experimental and in-progress.  We are building it up starting from "root"
-targets (that is, targets without dependencies), and expanding support to the rest of the library.
+There are two ways to include the Au library in your CMake project.
 
-To build the library using this experimental CMake support, follow these steps:
+1. **(Recommended)** Use the `FetchContent` module to download the library directly from GitHub.
 
-```sh
-# CMake is a "meta build system", not a build system.
-# This first command generates the actual build files.
-cmake -S . -B cmake/build -DCMAKE_VERIFY_INTERFACE_HEADER_SETS=TRUE
+2. Install the library to the system, and use `find_package`.
 
-# This command builds Au, checks include paths, and runs unit tests.
-cmake \
-  --build cmake/build \
-  --target \
-    all \
-    all_verify_interface_header_sets \
-    test
-```
+The reason we recommend `FetchContent` is because it makes it easier for each project to get the
+exact version of Au they need.  `FetchContent` also means you don't need to manually clone the Au
+repo, or build and run the tests.  On the other hand, if you want a single global system-wide
+version of Au, then you can install it to the system, and simply use `find_package`.
 
-#### Other build systems (CMake / conan / vcpkg / ...)
+In either case, here are the main targets and include files provided by the Au library:
 
-We would like to support all these build and packaging systems, and perhaps others!  But the initial
-public release is bazel-only, because bazel is what we use at Aurora, and we don't have experience
-with any of these alternatives.  Thus, we'll need to lean on the community to support them.
+| Target | Headers provided | Notes |
+|--------|------------------|-------|
+| `Au::au` | `"au/au.hh"`<br>`"au/units/*.hh"` | Core library functionality.  See [all available units](https://github.com/aurora-opensource/au/tree/main/au/units) |
+| `Au::io` | `"au/io.hh"` | `operator<<` support |
+| `Au::testing` | `"au/testing.hh"` | Utilities for writing googletest tests |
 
-Meanwhile, the library itself is still at least partially available on all build environments, via
-the single-file options explained above.
+Now, here are the installation instructions.
+
+!!! note
+    These instructions are for adding Au to a _project_ that uses CMake, not building Au itself
+    using CMake.
+
+    Au is a bazel-first project, so most Au development would normally be done using bazel.
+    However, we do have instructions for doing this with CMake as well: see the `find_package` tab
+    below.
+
+=== "Using `FetchContent`"
+    Add the following to your `CMakeLists.txt` file:
+
+    ```cmake
+    include(FetchContent)
+    FetchContent_Declare(
+      Au
+      GIT_REPOSITORY https://github.com/aurora-opensource/au
+      GIT_TAG "main"  # Or a specific tag.
+      EXCLUDE_FROM_ALL
+    )
+    FetchContent_MakeAvailable(Au)
+    ```
+
+    You should now be able to depend on Au targets, such as `Au::au` or `Au::io`, and include their
+    corresponding headers, such as `#include "au/au.hh"` or `#include "au/io.hh"`.
+
+=== "Using `find_package`"
+    Before you can use `find_package`, you need to install the library to your system.  This means
+    cloning the repo, building the library, running the tests, and installing it.
+
+    First, clone the repository.
+
+    ```sh
+    git clone https://github.com/aurora-opensource/au.git
+    cd au
+    ```
+
+    If you want a specific release, check out the tag you want.  Note that the first version of Au
+    that supports CMake will be 0.3.5.
+
+    ```sh
+    # Optional, but recommended:
+    git checkout "0.3.5"  # Or whichever tag you prefer.
+    ```
+
+    Now, build and test the library.  These commands will include both the explicit tests, and also
+    several CMake-generated tests to make sure that the includes are set up correctly.
+
+    ```sh
+    # CMake is a "meta build system", not a build system.
+    # This first command generates the actual build files.
+    cmake -S . -B cmake/build -DCMAKE_VERIFY_INTERFACE_HEADER_SETS=TRUE
+
+    # This command builds Au, checks include paths, and runs unit tests.
+    cmake \
+      --build cmake/build \
+      --target \
+        all \
+        all_verify_interface_header_sets \
+        test
+    ```
+
+    If the tests pass, you can install the library to your system.
+
+    ```sh
+    sudo cmake --install cmake/build
+    ```
+
+    At this point, the `Au` CMake library is installed to your system, and can be found via the
+    usual `find_package` mechanism!
+
+    ```cmake
+    find_package(Au)
+    ```
 
 <script src="../assets/hrh4.js" async=false defer=false></script>

